pkg/cri/sbserver/helpers_linux.go

```diff
@@ -65,6 +65,18 @@ func getCgroupsPath(cgroupsParent, id string) string {
 	return filepath.Join(cgroupsParent, id)
 }
 
+// getSandboxRootDir returns the root directory for managing sandbox files,
+// e.g. hosts files.
+func (c *criService) getSandboxRootDir(id string) string {
```

Comment:
Is there any reason here? it looks like it can be called in other platform

Reply:
Thanks, @fuweid.
It 's only used in `Linux`. Use `grep getSandboxRootDir` to get the result in `sbserver` directory.

```
podsandbox/sandbox_run.go
150:    sandboxRootDir := c.getSandboxRootDir(id)

podsandbox/sandbox_delete.go
41:     sandboxRootDir := c.getSandboxRootDir(sandboxID)

podsandbox/helpers.go
77:// getSandboxRootDir returns the root directory for managing sandbox files,
79:func (c *Controller) getSandboxRootDir(id string) string {

podsandbox/helpers_linux.go
70:     return filepath.Join(c.getSandboxRootDir(id), "hostname")
75:     return filepath.Join(c.getSandboxRootDir(id), "hosts")
80:     return filepath.Join(c.getSandboxRootDir(id), "resolv.conf")

helpers_linux.go
68:// getSandboxRootDir returns the root directory for managing sandbox files,
70:func (c *criService) getSandboxRootDir(id string) string {
82:     return filepath.Join(c.getSandboxRootDir(id), "hostname")
87:     return filepath.Join(c.getSandboxRootDir(id), "hosts")
92:     return filepath.Join(c.getSandboxRootDir(id), "resolv.conf")
```

Without this modification, the 
[Linters (macos-12)](https://github.com/containerd/containerd/actions/runs/3157416766/jobs/5138263894#logs) will fail.
```
run golangci-lint
  Running [/Users/runner/golangci-lint-1.48.0-darwin-amd64/golangci-lint run --out-format=github-actions --timeout=8m] in [] ...
  Error: func `(*criService).getVolatileSandboxRootDir` is unused (unused)
  Error: func `(*criService).getSandboxRootDir` is unused (unused)
  
  level=warning msg="[linters context] structcheck is disabled because of generics. You can track the evolution of the generics support by following the https://github.com/golangci/golangci-lint/issues/2649."
  
  Error: issues found
  Ran golangci-lint in 95[24](https://github.com/containerd/containerd/actions/runs/3157416766/jobs/5138263894#step:4:26)1ms
```

Reply:
Thanks, @fuweid.
It 's only used in `Linux`. Use `grep getSandboxRootDir` to get the result in `sbserver` directory.

```
podsandbox/sandbox_run.go
150:    sandboxRootDir := c.getSandboxRootDir(id)

podsandbox/sandbox_delete.go
41:     sandboxRootDir := c.getSandboxRootDir(sandboxID)

podsandbox/helpers.go
77:// getSandboxRootDir returns the root directory for managing sandbox files,
79:func (c *Controller) getSandboxRootDir(id string) string {

podsandbox/helpers_linux.go
70:     return filepath.Join(c.getSandboxRootDir(id), "hostname")
75:     return filepath.Join(c.getSandboxRootDir(id), "hosts")
80:     return filepath.Join(c.getSandboxRootDir(id), "resolv.conf")

helpers_linux.go
68:// getSandboxRootDir returns the root directory for managing sandbox files,
70:func (c *criService) getSandboxRootDir(id string) string {
82:     return filepath.Join(c.getSandboxRootDir(id), "hostname")
87:     return filepath.Join(c.getSandboxRootDir(id), "hosts")
92:     return filepath.Join(c.getSandboxRootDir(id), "resolv.conf")
```

Without this modification, the 
[Linters (macos-12)](https://github.com/containerd/containerd/actions/runs/3157416766/jobs/5138263894#logs) will fail.
```
run golangci-lint
  Running [/Users/runner/golangci-lint-1.48.0-darwin-amd64/golangci-lint run --out-format=github-actions --timeout=8m] in [] ...
  Error: func `(*criService).getVolatileSandboxRootDir` is unused (unused)
  Error: func `(*criService).getSandboxRootDir` is unused (unused)
  
  level=warning msg="[linters context] structcheck is disabled because of generics. You can track the evolution of the generics support by following the https://github.com/golangci/golangci-lint/issues/2649."
  
  Error: issues found
  Ran golangci-lint in 95[24](https://github.com/containerd/containerd/actions/runs/3157416766/jobs/5138263894#step:4:26)1ms
```

Reply:
I think we have a bit of code duplication between `sbserver/helpers` and `sbserver/podsandbox/helpers`. I'll do some cleaning afterwards.

Reply:
@fuweid your grep seems to show it also being used from podsandbox/sandbox_run.go:150 and podsandbox/sandbox_delete.go:41

Reply:
> @fuweid your grep seems to show it also being used from podsandbox/sandbox_run.go:150 and podsandbox/sandbox_delete.go:41

Yes, but only Linux. so move it to *_linux.go file.

Reply:
sorry for the late reply. Let's remove the duplication in the followup